src/Symfony/Component/Notifier/Bridge/LightSms/LightSmsTransport.php

```diff
@@ -0,0 +1,162 @@
+        '24' => 'Timestamp not specified',
+        '25' => 'Error in access to the list',
+        '26' => 'There are no numbers in the list',
+        '27' => 'No valid numbers',
+        '28' => 'Date of start not specified (Format: YYYY-MM-DD)',
+        '29' => 'Date of end not specified (Format: YYYY-MM-DD)',
+        '30' => 'No date (format: YYYY-MM-DD)',
+        '31' => 'Closing direction to the user',
+        '32' => 'Not enough money',
+        '33' => 'Phone number is not set',
+        '34' => 'Phone is in stop list',
+        '35' => 'Not enough money',
+        '36' => 'Can not obtain information about phone',
+        '37' => 'Base Id is not set',
+        '38' => 'Phone number is already exist in this base',
+        '39' => 'Phone number is not exist in this base',
+    ];
+
+    public function __construct(
+        string $login,
+        string $password,
+        string $phone,
+        HttpClientInterface $client = null,
+        EventDispatcherInterface $dispatcher = null
+    ) {
+        $this->login = $login;
+        $this->password = $password;
+        $this->phone = $phone;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('lightsms://%s?phone=%s', $this->getEndpoint(), $this->phone);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage;
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, SmsMessage::class, $message);
+        }
+
+        $this->message = $message;
+
+        $signature = $this->getSignature();
+
+        $endpoint = sprintf(
+            'https://%s?login=%s&signature=%s&phone=%s&text=%s&sender=%s&timestamp=%s',
+            $this->getEndpoint(),
+            $this->login,
+            $signature,
+            str_replace('+', '', $message->getPhone()),
+            $message->getSubject(),
+            $this->phone,
+            time()
+        );
+
+        $response = $this->client->request('GET', $endpoint);
+
+        $content = $response->toArray(false);
+
+        if (isset($content['error'])) {
+            throw new TransportException('Unable to send the SMS: '.$this->errorCodes[$content['error']], $response);
+        }
+
+        $phone = preg_replace("/[^\d]/", '', $message->getPhone());
+        if (32 == $content[$phone]['error']) {
+            throw new TransportException('Unable to send the SMS: '.$this->errorCodes[$content['error']], $response);
+        }
+
+        if (0 == $content[$phone]['error']) {
+            $sentMessage = new SentMessage($message, (string) $this);
+            $sentMessage->setMessageId($content[$phone]['id_sms']);
+        }
+
+        return $sentMessage;
+    }
+
+    private function getSignature(): string
```

Comment:
maybe pass the message as an argument instead of setting it as a property?

```suggestion
    private function getSignature(SmsMessage $message): string
```

Reply:
Yes, you are right. Fixed.